versionedRouter.js

```diff
@@ -95,34 +96,70 @@ const functionsEnabled = () => {
 };
 
 async function handleCdkV2(destName, parsedEvent, flowType) {
-  try {
-    const workflowEngine = await getWorkflowEngine(destName, flowType);
+  const workflowEngine = await getWorkflowEngine(destName, flowType);
 
-    const result = await workflowEngine.execute(parsedEvent);
-    // TODO: Handle remaining output scenarios
-    return result.output;
-  } catch (err) {
-    const errorInfo = getErrorInfo(err);
+  const result = await workflowEngine.execute(parsedEvent);
+  // TODO: Handle remaining output scenarios
+  return result.output;
+}
 
-    // TODO: Bump the error priority even further as it's an unhandled error in the CDK
-    const errObj = generateErrorObject(
-      errorInfo,
-      destName,
-      TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM
-    );
+async function getCdkV2Result(destName, event, flowType) {
+  const cdkResult = {};
+  try {
+    cdkResult.output = await handleCdkV2(destName, event, flowType);
+  } catch (error) {
+    cdkResult.error = {
+      message: error.message,
+      statusCode: getErrorStatusCode(error)
+    };
+  }
+  return cdkResult;
+}
 
-    // Dump the raw error
-    logger.error(err);
+async function compareWithCdkV2(destType, input, flowType, v0Result) {
+  try {
+    const cdkResult = await getCdkV2Result(destType, input, flowType);
+    const unmatchedKeys = Object.keys(
+      CommonUtils.objectDiff(v0Result, cdkResult)
+    );
+    if (unmatchedKeys.length > 0) {
+      logger.error(
+        `[LIVE_COMPARE_TEST] failed for destType=${destType}, flowType=${flowType}, unmatchedKeys=${unmatchedKeys}, metadata=${JSON.stringify(
+          input.metadata
+        )}`
+      );
+      return;
+    }
+    logger.debug(
```

Comment:
These logs will be overwhelming whenever we enable them. It's better to remove completely or sample them at a decent rate.

Reply:
Sure, that is for testing, we can remove

Reply:
Done.